core/jobs/batch_jobs/skill_validation_jobs.py

```diff
@@ -0,0 +1,136 @@
+            or at least one explanation exceeds 300 characters.
+        """
+        if len(explanations) > 10:
+            return True
+
+        for explanation in explanations:
+            if len(explanation) > 300:
+                return True
+
+        return False
+
+    def get_count_of_rubrics_with_invalid_explanations(self, skill):
+        """Returns the count of rubrics with invalid explanations in a skill.
+
+        Returns: int. The count of rubrics with invalid explanations.
+        """
+        count = 0
+        rubrics = skill[1]
+        for rubric in rubrics:
+            if self.rubric_explanations_are_invalid(rubric['explanations']):
+                count = count + 1
+        return count
+
+    def filter_skills_having_rubrics_with_invalid_explanations(self, skill):
+        """Returns True if skill has rubrics with invalid explanation.
+
+        Returns:
+            bool. Returns True if skill has rubrics with invalid explanation.
+        """
+        rubrics = skill[1]
+        for rubric in rubrics:
+            if self.rubric_explanations_are_invalid(rubric['explanations']):
+                return True
+        return False
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of skills with invalid rubric explanations.
+
+        Returns:
+            PCollection. Returns a PCollection of skills with invalid
+            rubric explanations.
+        """
+        total_skills = (
+            self.pipeline
+            | 'Get all SkillModels' >> ndb_io.GetModels(
+                skill_models.SkillModel.get_all(include_deleted=False))
+            | 'Combine skill ids and rubrics' >> beam.Map(
+                lambda skillModel: (
+                    skillModel.id, skillModel.rubrics))
+        )
+
+        skills_having_rubrics_with_invalid_explanation = (
+            total_skills
+            | 'Filter skills with invalid rubrics explanation' >> beam.Filter(
+                self.filter_skills_having_rubrics_with_invalid_explanations)
+        )
+
+        report_number_of_skills_queried = (
+            total_skills
+            | 'Report count of skill models' >> (
+                job_result_transforms.CountObjectsToJobRunResult('SKILLS'))
+        )
+
+        report_number_of_skills_with_invalid_rubric_explanations = (
+            skills_having_rubrics_with_invalid_explanation
+            | 'Report count of skills with invalid rubric explanations' >> (
+                job_result_transforms.CountObjectsToJobRunResult('INVALID'))
+        )
+
+        report_invalid_skill_ids_and_rubrics = (
+            skills_having_rubrics_with_invalid_explanation
+            | 'Report invalid skill ids and rubrics' >> (
+              beam.Map(
+                lambda skill: job_run_result.JobRunResult.as_stderr(
+                    'The id of the skill is %s and number of '
+                    'invalid rubrics are %s' % (
```

Comment:
I think we also need to output some detail about the skill, otherwise, we will not be able to locate it. We also need to deal with the errors right? We will need to see the skill which is invalid so that could be fixed later.
Thanks

Reply:
Can we show the difficulty of the rubrics? Because skills can have maximum of 3 rubrics (one for each difficulty). If we have the difficulty, then we can locate the rubrics easily.

Reply:
I think we should be very clear of which particular rubric explanation is invalid. `Because skills can have maximum of 3 rubrics (one for each difficulty)` I'm not sure about this line because as far as I know, we can have almost 10 explanations of each difficulty and I guess each explanation length can be maximum of 300 characters.

Reply:
So it will be good if we can specify that `the easy rubric explanation have {{explanation}} invalid`, something like this.

Reply:
done